### PR TITLE
fix: tweaking relative path handling for windows compatibility

### PR DESCRIPTION
**Pull request by aorumbayev**
## Proposed changes

1. Fixing path resolution for windows paths in resolveFilePath method
2. Adding shx to make compile script available on windows as well
3. Adding tweaks to esbuild to target node since fix relies on `path`. @jasonpaulos given that its a vscode extension related core dependency - safe to assume we can target node? I think all electron apps got node available

### CODE REVIEW

**Comment by aorumbayev** [comment]
@jasonpaulos would appreciate if you could trigger a new deployment to npm once this is reviewed and merged

File: package.json
Comment (jasonpaulos) [inline package.json:59]: Please revert this, this is meant to build for the browser

**Review by jasonpaulos — COMMENTED** [review]
I can help figure out the best way to support windows paths in the debug adapter, but it will be much easier if we have reliable testing on windows. Since this is an area you've shown interest in, would you be willing to modify the github actions workflow to also build and test on windows?

**Comment by aorumbayev** [comment]
Thanks @jasonpaulos. I pushed a tweak to ci to include windows worker, will check the remaining comments tomorrow

File: package.json
Comment (robdmoore) [reply package.json:59]: Looks like it's the same as before but moved up in the file so it's next to the base build AFAICT

File: package.json
Comment (robdmoore) [inline package.json:101]: TIL! I usually just use `rimraf`, `copyfiles` etc. directly, but this is neat!

File: src/common/utils.ts
Comment (robdmoore) [inline src/common/utils.ts:137]: You can probably skip this and just use the output from the URL object? forward slashes do work on windows, then we don't have the browser incompatibility problem Jason mentioned

File: package.json
Comment (neilcampbell) [reply package.json:59]: I'm suprised this actually works in the browser, given the `AvmRuntime` extends `EventEmitter`.

File: src/common/utils.ts
Comment (aorumbayev) [reply src/common/utils.ts]: Using Rob's suggestion instead, using URL pathname was still enough, same effect even if i remove the path.resolve

File: package.json
Comment (aorumbayev) [reply package.json]: Reverted

File: .github/workflows/ci.yml
Comment (jasonpaulos) [inline .github/workflows/ci.yml:13]: Looks like prettier is failing on the windows build. I found this comment, which suggests line endings may be the issue: https://github.com/prettier/prettier/issues/7825

Perhaps we should add a `.gitattributes` file like it suggests?

**Review by jasonpaulos — COMMENTED** [review]
Ok sounds good. I'll just leave these two comments with some insights from my end

File: src/common/utils.ts
Comment (jasonpaulos) [inline src/common/utils.ts:130]: This comment suggests an alternative way of creating the URL that might be more windows friendly: https://github.com/nodejs/node/issues/20080#issuecomment-381686455


```javascript
new URL(`file:///${path}`)
```

If that works, it'd be nice to avoid manually adjusting the file separator

**Comment by jasonpaulos** [comment]
You should be able to run the workflow without my approval now

**Comment by jasonpaulos** [comment]
I hope you don't mind but I pushed some changes to this PR. I liked the idea of expanding `FileAccessor` to implement this function, but there were a few things I wanted to tweak. Please let me know if this looks good to you @aorumbayev and @neilcampbell.

Specifically, if you have access to a windows machine, can you please run the extension and ensure the `FileAccessor` for vscode is implemented properly? Since that relies on the vscode runtime package I unfortunately can't think of a way to automate that test.

**Comment by aorumbayev** [comment]
@jasonpaulos awesome, thanks for pushing the tweaks. Smoke tested this on windows independently and as a dependency in our extension repo - works fine 👍

I can't self approve the repo so feel free to merge as time allows.

**Review by neilcampbell — APPROVED** [review]

**Pull request merged**
